corehq/apps/geospatial/management/commands/index_geolocation_case_properties.py

```diff
@@ -25,64 +14,26 @@ class Command(BaseCommand):
     def add_arguments(self, parser):
         parser.add_argument('domain')
         parser.add_argument('--case_type', required=False)
-        parser.add_argument('--query_limit', required=False)
-        parser.add_argument('--chunk_size', required=False)
 
     def handle(self, *args, **options):
         domain = options['domain']
         case_type = options.get('case_type')
-        query_limit = options.get('query_limit')
-        chunk_size = options.get('chunk_size')
-        index_case_docs(domain, query_limit, chunk_size, case_type)
+        index_case_docs(domain, case_type)
 
 
-def index_case_docs(domain, query_limit=DEFAULT_QUERY_LIMIT, chunk_size=DEFAULT_CHUNK_SIZE, case_type=None):
+def index_case_docs(domain, case_type=None):
     geo_case_property = get_geo_case_property(domain)
-    query = _es_case_query(domain, geo_case_property, case_type)
+    query = case_query_for_missing_geopoint_val(domain, geo_case_property, case_type)
     count = query.count()
     print(f'{count} case(s) to process')
-    batch_count = 1
-    if query_limit:
-        batch_count = math.ceil(count / query_limit)
+    batch_count = math.ceil(count / DEFAULT_QUERY_LIMIT)
     print(f"Cases will be processed in {batch_count} batches")
     for i in range(batch_count):
         print(f'Processing {i+1}/{batch_count}')
-        query = _es_case_query(domain, geo_case_property, case_type, size=query_limit)
-        case_ids = query.get_ids()
-        _index_case_ids(domain, case_ids, chunk_size)
-
-
-def _index_case_ids(domain, case_ids, chunk_size):
-    for case_id_chunk in chunked(with_progress_bar(case_ids), chunk_size):
-        case_chunk = CommCareCase.objects.get_cases(list(case_id_chunk), domain)
-        case_search_adapter.bulk_index(case_chunk)
-    manager.index_refresh(case_search_adapter.index_name)
-
-
-def _es_case_query(domain, geo_case_property, case_type=None, size=None):
-    query = (
-        CaseSearchES()
-        .domain(domain)
-        .filter(_geopoint_value_missing_for_property(geo_case_property))
-    )
-    if case_type:
-        query = query.case_type(case_type)
-    if size:
-        query = query.size(size)
-    return query
-
-
-def _geopoint_value_missing_for_property(geo_case_property_name):
-    """
-    Query to find docs with missing 'geopoint_value' for the given case property.
-    """
-    return queries.nested(
-        CASE_PROPERTIES_PATH,
-        queries.filtered(
-            queries.match_all(),
-            filters.AND(
-                filters.term(PROPERTY_KEY, geo_case_property_name),
-                filters.missing(PROPERTY_GEOPOINT_VALUE)
-            )
+        process_batch(
+            domain,
+            geo_case_property,
+            total_count=count,
+            case_type=case_type,
+            with_progress=True,
```

Comment:
We probably need offset here as well after the change to not refresh the index with each batch.

Reply:
Definitely, good catch. Addressed in [3b96a19](https://github.com/dimagi/commcare-hq/pull/35126/commits/3b96a19999d2f2a27bd45f8e2939245531845bf9).